src/markdown.c

```diff
@@ -769,8 +769,33 @@ char_entity(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t max_
 		if (!is_valid_numeric_entity(entity_val))
 			return 0;
 	} else {
-		if (!is_allowed_named_entity((const char *)data, end))
+		/* Lookup the entity in the named entity table. */
+		const struct html_entity *entity = is_allowed_named_entity((const char *)data, end);
+		if (!entity)
 			return 0;
+		if (entity->codepoints[0]) {
+			/* Convert the entity to numeric entities. */
+			size_t entitystr_size = MAX_NUM_ENTITY_LEN+10;
+			char entitystr[entitystr_size];
+			int i;
+			for (i = 0; i < MAX_ENTITY_CODEPOINTS && entity->codepoints[i]; i++) {
+				/* Verify the codepoint is a valid entity. */
+				int entitystr_len;
+				entity_val = entity->codepoints[i];
+				assert(is_valid_numeric_entity(entity_val));
+				/* Render codepoint to an entity. */
+				entitystr_len = snprintf(entitystr, entitystr_size, "&#x%X;", entity_val);
+				assert(entitystr_len < entitystr_size);
```

Comment:
Hmm, in other places I'd do something like

```c
if (entitystr_len >= entitystr_size) {
    assert(0);
    return;
}
```

but that could silently leave the output buffer with just one codepoint appended... I'm not sure what standard practice is for handling fatal errors is in production C these days. Printing to stderr and intentionally dereferencing `NULL`? @spladug Any preference here?